web/src/main/java/org/cbioportal/web/StudyViewController.java

```diff
@@ -81,8 +81,16 @@
         List<String> studyIds = new ArrayList<>();
         List<String> sampleIds = new ArrayList<>();
         studyViewFilterUtil.extractStudyAndSampleIds(filteredSampleIdentifiers, studyIds, sampleIds);
-        return new ResponseEntity<>(clinicalDataService.fetchClinicalDataCounts(studyIds, sampleIds, 
-            Arrays.asList(attributeId), clinicalDataType.name()).get(attributeId), HttpStatus.OK);
+        Map<String, List<ClinicalDataCount>> resultForSampleAttributes = clinicalDataService.fetchClinicalDataCounts(
+            studyIds, sampleIds, attributes.stream().filter(a -> a.getClinicalDataType().equals(ClinicalDataType.SAMPLE))
+            .map(a -> a.getAttributeId()).collect(Collectors.toList()), ClinicalDataType.SAMPLE.name());
+        Map<String, List<ClinicalDataCount>> resultForPatientAttributes = clinicalDataService.fetchClinicalDataCounts(studyIds, sampleIds, 
+        attributes.stream().filter(a -> a.getClinicalDataType().equals(ClinicalDataType.PATIENT))
+            .map(a -> a.getAttributeId()).collect(Collectors.toList()), ClinicalDataType.PATIENT.name());
+        Map<String, List<ClinicalDataCount>> combinedResult = new HashMap<>();
+        combinedResult.putAll(resultForSampleAttributes);
```

Comment:
there might be a possibility for same key in `resultForSampleAttributes` and `resultForPatientAttributes`

Reply:
actually attribute id is unique for a cancer study in clinical_attribute_meta, so they can't be the same. I just require the clinicalDataType from the frontend so that we wouldn't need to check the clinical_attribute_meta table first and then use the appropriate table when getting clinical data, because frontend already has that information

Reply:
Multi-study queries actually can have a sample attribute in one study and the a patient attribute with the same id in another study, so I'll change the response a little bit.